manifests/manage_app_pool.pp

```diff
@@ -156,6 +157,18 @@
       require   => Exec["Create-${app_pool_name}"],
       logoutput => true,
     }
+    
+    
+if $apppool_idle_Timeout_action {
+    validate_re($apppool_idle_Timeout_action, '^(Suspend|Terminate)$')
+    exec { "IdleTimeoutAction-${app_pool_name}":
+      command   => "Import-Module WebAdministration; Set-ItemProperty \"IIS:\\AppPools\\${app_pool_name}\" startMode ${apppool_idle_Timeout_action}",
+      provider  => powershell,
+      onlyif    => "Import-Module WebAdministration; if((Get-ItemProperty \"IIS:\\AppPools\\${app_pool_name}\" startMode).CompareTo('${apppool_idle_Timeout_action}') -eq 0) { exit 1 } else { exit 0 }",
```

Comment:
Since you've got this `if`, should the new parameter default to `undef` instead of `'Terminate'`?